item-impl/src/main/java/com/example/auction/item/impl/ItemRepository.java

```diff
@@ -0,0 +1,248 @@
+        private PreparedStatement insertItemSummaryByCreatorStatement;
+        private PreparedStatement updateItemSummaryStatusStatement;
+
+        @Inject
+        public PItemEventProcessor(CassandraSession session, CassandraReadSide readSide) {
+            this.session = session;
+            this.readSide = readSide;
+        }
+
+        @Override
+        public ReadSideHandler<PItemEvent> buildHandler() {
+            return readSide.<PItemEvent>builder("pItemEventOffset")
+                    .setGlobalPrepare(this::createTables)
+                    .setPrepare(tag -> prepareStatements())
+                    .setEventHandler(PItemEvent.ItemCreated.class,
+                            e -> insertItem(e.getItem()))
+                    .setEventHandler(PItemEvent.AuctionStarted.class,
+                            e -> updateItemSummaryStatus(e.getItemId(), ItemStatus.AUCTION))
+                    .setEventHandler(PItemEvent.AuctionFinished.class,
+                            e -> updateItemSummaryStatus(e.getItemId(), ItemStatus.COMPLETED))
+                    .build();
+        }
+
+        @Override
+        public PSequence<AggregateEventTag<PItemEvent>> aggregateTags() {
+            return PItemEvent.TAG.allTags();
+        }
+
+        private CompletionStage<Done> createTables() {
+            return doAll(
+                    session.executeCreateTable(
+                            "CREATE TABLE IF NOT EXISTS itemCreator (" +
+                                    "itemId timeuuid PRIMARY KEY, " +
+                                    "creatorId UUID" +
+                                    ")"
+                    ),
+                    session.executeCreateTable(
+                            "CREATE TABLE IF NOT EXISTS itemSummaryByCreator (" +
+                                    "creatorId UUID, " +
+                                    "itemId timeuuid, " +
+                                    "title text, " +
+                                    "currencyId text, " +
+                                    "reservePrice int, " +
+                                    "status text, " +
+                                    "PRIMARY KEY (creatorId, itemId) " +
+                                    ") " +
+                                    "WITH CLUSTERING ORDER BY (itemId DESC)"
+                    ).thenCompose(done ->
+                            session.executeCreateTable(
+                                    "CREATE MATERIALIZED VIEW IF NOT EXISTS itemSummaryByCreatorAndStatus AS " +
+                                            "SELECT * FROM itemSummaryByCreator " +
+                                            "WHERE status IS NOT NULL AND itemId IS NOT NULL " +
+                                            "PRIMARY KEY (creatorId, status, itemId) " +
+                                            "WITH CLUSTERING ORDER BY (status ASC, itemId DESC)"
+                            )
+                    )
```

Comment:
A subtlety here is that the two table creation statements can execute potentially in parallel or in an arbitrary order because there's no dependency between them, but the materialized view does depend on the second table creation.